introspection-engine/connectors/sql-introspection-connector/src/introspection.rs

```diff
@@ -239,3 +230,35 @@ fn calculate_fields_for_prisma_join_table(
         }
     }
 }
+
+fn introspect_enums(datamodel: &mut Datamodel, ctx: &Context) {
+    let existing_enums_by_database_name: HashMap<&str, _> = ctx
+        .previous_schema()
+        .db
+        .walk_enums()
+        .map(|enm| (enm.database_name(), enm.id))
+        .collect();
+
+    datamodel.enums = ctx
+        .schema
+        .enum_walkers()
+        .map(|enm| sql_enum_to_dml_enum(enm, ctx))
```

Comment:
I wonder, when we remove DML from the introspection, what would the conversion be? Or more like `Pair<Option<psl::Enum>, Option<describer::Enum>>`, which will then give us the answer what is the correct name and mapping, etc?

Reply:
I think we could probably already push something into the renderer, for enums. For models there will probably need to be some intermediate steps because of relations.